src/Domain/CoordinateMaps/CoordinateMap.tpp

```diff
@@ -448,35 +449,56 @@ auto CoordinateMap<SourceFrame, TargetFrame, Maps...>::
   tuple_transform(
       maps_,
       [&frame_velocity, &jac, &mapped_point, time, &functions_of_time](
-          const auto& map, auto index, const std::tuple<Maps...>& maps) {
+          const auto& map, auto index, const std::tuple<Maps...>& /*maps*/) {
         constexpr size_t count = decltype(index)::value;
         using Map = std::decay_t<decltype(map)>;
+        static constexpr bool is_time_dependent =
+            domain::is_map_time_dependent_v<Map>;
+        static constexpr bool use_combined_shape_call =
+            std::is_same_v<CoordinateMaps::TimeDependent::Shape, Map> and
+            std::is_same_v<T, DataVector>;
```

Comment:
Instead of checking if this is a shape map explicitly (and thus adding an strange dependency for the general coord map on the time dependent maps), add a meta function that checks if the `Map` object has a function named `coords_frame_velocity_jacobian` that can be called with the given arguments.

Then this easily allows us to edit the other time dependent maps if we want to to add this feature to those as well

Reply:
Note: I think we could get away with a forward declaration of the map to avoid the coupling. Just providing another option.